### PR TITLE
[Core] Testing new algorithm versus previous one for the nearest point in a triangle

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

image

New one it is not just faster, also provides a smaller distance (the point of the method).

**🆕 Changelog**

- New tests for method

### CODE REVIEW

File: kratos/tests/cpp_tests/utilities/test_nearest_point_utilities.cpp [hunk 1]
```diff
@@ -20,8 +21,96 @@
...
+ *     5    /       4      \    7
+ *         /                \
+ *        /                  \
+ * @tparam Type of the Point
+ * @tparam TGeometryType The type of the triangle. Assumes to have [] access and IsInside method
+ * @param rPoint The query point which we want to get nearest point to it on the line
+ * @param rTriangle The triangle in which we want to find the nearest point to rPoint
+ * @return The nearest point to rPoint
+ */
+template<class TPointType, class TGeometryType>
```
Comment (rubenzorrilla) [inline kratos/tests/cpp_tests/utilities/test_nearest_point_utilities.cpp:75]: I'd put this into an anonymous namespace so we completely avoid using it outside the context of this test.

**Review by rubenzorrilla — COMMENTED** [review]
Just for me to understand, if current implementation is not only faster but more accurate, which is the point of testing against previous implementation?

**Comment by rubenzorrilla** [comment]
> **📝 Description**
> 
> image
> 
> New one it is not just faster, also provides a smaller distance (the point of the method).
> 
> **🆕 Changelog**
> 
> * New tests for method

Which is the location of the origin point? (curiosity)

**Comment by loumalouomega** [comment]
> > **📝 Description**
> > image
> > New one it is not just faster, also provides a smaller distance (the point of the method).
> > **🆕 Changelog**
> > 
> > * New tests for method
> 
> Which is the location of the origin point? (curiosity)

The corner cases I got it from a real industrial case.

**Comment by loumalouomega** [comment]
> Just for me to understand, if current implementation is not only faster but more accurate, which is the point of testing against previous implementation?

I wanted to prove that it is. Okay, I can remove this.

**Pull request closed without merging**